src/plugins/data/public/plugin.ts

```diff
@@ -168,6 +170,7 @@ export class DataPublicPlugin
 
     const ac = this.autocomplete.setup(core);
     ac.addQuerySuggestionProvider('SQL', getSQLSuggestions);
+    ac.addQuerySuggestionProvider('kuery', getDQLSuggestions);
```

Comment:
Since the autocomplete service is initialized here, rather than create a getter/setter for `http`, we could just use it from the autocomplete service, since it gets set up with it.
```
const ac = this.autocomplete.setup(core);
```

Reply:
Done. I pass the core setup along with the rest of the arguments whenever a query suggestion provider is called